gfx-descriptor/src/counts.rs

```diff
@@ -175,6 +175,11 @@ impl DescriptorCounts {
         counts: [0; DESCRIPTOR_TYPES_COUNT],
     };
 
+    /// Returns true if this indicates a completely empty descriptor layout.
+    pub fn is_empty(&self) -> bool {
+        self.counts == Self::EMPTY.counts
```

Comment:
Even though we `derive(Eq, PartialEq)` on the `DescriptorCounts` type, Rust doesn't allow me to use `==` between two instances of this type. I think it might be because it contains a slice of a fixed size and Rust doesn't have equality implemented for it.